nautobot/extras/api/views.py

```diff
@@ -535,22 +530,16 @@ def _run_job(request, job_model, legacy_response=False):
         input_serializer.is_valid(raise_exception=True)
 
         data = input_serializer.validated_data.get("data", {})
-        commit = input_serializer.validated_data.get("commit", None)
         task_queue = input_serializer.validated_data.get("task_queue", default_valid_queue)
         schedule_data = input_serializer.validated_data.get("schedule", None)
 
-    if commit is None:
-        commit = job_model.commit_default
-
     if task_queue not in valid_queues:
         raise ValidationError({"task_queue": [f'"{task_queue}" is not a valid choice.']})
 
     cleaned_data = None
     try:
-        cleaned_data = job.validate_data(data, files=files)
-        cleaned_data.pop(
-            "_commit", None
-        )  # We don't get commit from the form, instead it's part of the serializer's validated data
+        cleaned_data = job_class().validate_data(data, files=files)
+        cleaned_data = job_model.job_class.prepare_job_kwargs(cleaned_data)
```

Comment:
Feels weird from an API standpoint that `validate_data()` is an instance method but `prepare_job_kwargs` is a classmethod?

Reply:
`validate_data` should be a class method. The only things it's accessing on `self` are class properties (`task_queues` and `field_order`)